fix(docker): remove manual network creation causing Compose label mismatch

## Problem

`make up` fails on macOS (Docker Desktop) with:

```
network goclaw-net was found but has incorrect label com.docker.compose.network set to "" (expected: "goclaw-net")
make: *** [up] Error 1
```

## Root Cause

The Makefile `net` target (line 39-40) creates `goclaw-net` via `docker network create`, which does not set the `com.docker.compose.network` label. When Docker Compose later tries to use the same network, it rejects it due to the missing/incorrect label.

```makefile
net:
	docker network inspect goclaw-net >/dev/null 2>&1 || docker network create goclaw-net
```

Both `up` and `reset` depend on this target.

## Environment

- macOS (Docker Desktop)
- Docker Compose v2

## Proposed Fix

Remove the `net` target entirely. `docker-compose.yml` already defines `goclaw-net` as a Compose-managed network — Compose creates it automatically with the correct labels.

```diff
-.PHONY: build run clean version net up down logs reset test vet check-web dev migrate setup ci
+.PHONY: build run clean version up down logs reset test vet check-web dev migrate setup ci

-net:
-	docker network inspect goclaw-net >/dev/null 2>&1 || docker network create goclaw-net
-
-up: net version-file
+up: version-file

-reset: net version-file
+reset: version-file
```

Verified locally: after removing the stale network (`docker network rm goclaw-net`), `make up` proceeds without the label error.

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ VERSION ?= $(shell git describe --tags --abbrev=0 2>/dev/null || echo dev)
 LDFLAGS  = -s -w -X github.com/nextlevelbuilder/goclaw/cmd.Version=$(VERSION)
 BINARY   = goclaw
 
-.PHONY: build run clean version net up down logs reset test vet check-web dev migrate setup ci desktop-dev desktop-build desktop-dmg
+.PHONY: build run clean version up down logs reset test vet check-web dev migrate setup ci desktop-dev desktop-build desktop-dmg
 
 build:
 	CGO_ENABLED=0 go build -ldflags="$(LDFLAGS)" -o $(BINARY) .
@@ -36,13 +36,10 @@ endif
 COMPOSE = $(COMPOSE_BASE) $(COMPOSE_EXTRA)
 UPGRADE = docker compose -f docker-compose.yml -f docker-compose.postgres.yml -f docker-compose.upgrade.yml
 
-net:
-	docker network inspect goclaw-net >/dev/null 2>&1 || docker network create goclaw-net
-
 version-file:
 	@echo $(VERSION) > VERSION
 
-up: net version-file
+up: version-file
 	$(COMPOSE) up -d --build
 	$(UPGRADE) run --rm upgrade
 
@@ -52,7 +49,7 @@ down:
 logs:
 	$(COMPOSE) logs -f goclaw
 
-reset: net version-file
+reset: version-file
 	$(COMPOSE) down -v
 	$(COMPOSE) up -d --build
 

